File: data/src/main/java/com/arturogutierrez/openticator/domain/PreferencesImpl.java
package com.arturogutierrez.openticator.domain;

import android.content.SharedPreferences;
import javax.inject.Inject;

public class PreferencesImpl implements Preferences {

  private static final String MASTER_PASSWORD_ID = "MASTER_ID";

  private final SharedPreferences sharedPreferences;

  @Inject
  public PreferencesImpl(SharedPreferences sharedPreferences) {
    this.sharedPreferences = sharedPreferences;
  }

  @Override
  public void reset() {
    sharedPreferences.edit().clear().apply();
  }

  @Override
  public String getMasterPassword() {
    return sharedPreferences.getString(MASTER_PASSWORD_ID, null);
  }

  @Override
  public void setMasterPassword(String plainMasterPassword) {
    sharedPreferences.edit().putString(MASTER_PASSWORD_ID, plainMasterPassword).apply();
  }
}

File: data/src/main/kotlin/com/arturogutierrez/openticator/storage/preferences/PreferencesImpl.kt
package com.arturogutierrez.openticator.storage.preferences

import android.content.SharedPreferences
import javax.inject.Inject

class PreferencesImpl @Inject constructor(val sharedPreferences: SharedPreferences) : Preferences {

  private companion object {
    val MASTER_PASSWORD_ID = "MASTER_ID"
  }

  override fun reset() {
    sharedPreferences.edit().clear().apply()
  }

  override var masterPassword: String?
    get() = sharedPreferences.getString(MASTER_PASSWORD_ID, null)
    set(masterPassword) = sharedPreferences.edit().putString(MASTER_PASSWORD_ID, masterPassword).apply()
}
